### PR TITLE
BUG: Check complib values

### LINKED ISSUE #4582: ER: raise on invalid compression options in HDFStore
http://stackoverflow.com/questions/18274973/using-compression-with-pandas-and-hd5-hdfstore/18275402#18275402

### LINKED ISSUE #8874: HDF corrupts data when using complib='blosc:zlib'
I'm not sure if this is supported or not -- it isn't in the doc string for HDFStore, but it seems to be allowed by the HDFStore (nothing is raised).

Unfortunately so far I can only get it to show the bad behavior on a proprietary dataset, which is storing a `pd.Panel` which contains items of of mixed types.  

Some of the float values are be changed from small (|x|<1.0) to very large (3.xe+308).

Is the compressor just passed through to pytables? If so, this might be a pytables issue.

```
pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.8.final.0
python-bits: 64
OS: Linux
OS-release: 2.6.32-431.23.3.el6.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.15.1-24-g54e237b
nose: 1.3.4
Cython: 0.21.1
numpy: 1.9.1
scipy: 0.14.0
statsmodels: 0.7.0.dev-c8e980d
IPython: 2.3.1
sphinx: 1.2.3
patsy: 0.3.0
dateutil: 1.5
pytz: 2014.9
bottleneck: None
tables: 3.1.1
numexpr: 2.3.1
matplotlib: 1.4.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
rpy2: None
sqlalchemy: 0.9.8
pymysql: None
psycopg2: None
```

### PR DESCRIPTION
**Pull request by bashtage**
Add check for complib when opening a HDFStore

closes #4582
closes #8874

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -389,6 +389,10 @@ def __init__(self, path, mode=None, complevel=None, complib=None,
         except ImportError as ex:  # pragma: no cover
             raise ImportError('HDFStore requires PyTables, "{ex}" problem importing'.format(ex=str(ex)))
 
+        if complib not in (None, 'blosc', 'bzip2', 'lzo', 'zlib'):
```
Comment (jreback) [inline pandas/io/pytables.py]: I think this should iterrogate the ``PyTables/Filters`` class, as more things seem to be valid http://www.pytables.org/usersguide/libref/helper_classes.html#filtersclassdescr

Comment (jreback) [inline pandas/io/pytables.py]: but then reading over the old issue. I am not sure these ``blosc`` compressors work (though not entirely sure why)

**Comment by bashtage** [comment]
Using multiple compressors, e.g. `blosc:zlib`, resulted in corruption for me - also not sure why. So it seems safer to just prevent them being used and to enforce what the docs say are supported compressors.

**Comment by jreback** [comment]
ok that's fine, move to a class variable though ``_compressors``? to make maintenance easier in the future

**Comment by jreback** [comment]
ping when green. I am going to cut 0.16.2 soon, so this will be last one :)

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/3cdd1df36bdd7a49d629aaae70b65616ea702771

thanks!

**Pull request closed without merging**